### ISSUE TITLE
`redirect` does not support redirecting to nullable values

### ISSUE THREAD

**Issue by mnonnenmacher** [SMILEY4/schema-kenerator#14]
I have a class `OptionalValue<T>` which is unwrapped to `T` during serialization and use the `redirect` function to get the correct schema:

```kotlin
processReflection {
    redirect<OptionalValue<String>, String>()
    redirect<OptionalValue<String?>, String?>()
}
```

The first line with `OptionalValue<String>` is working fine. But for the second line with `OptionalValue<String?>` the nullability information is lost. I think the problem is here:

https://github.com/SMILEY4/schema-kenerator/blob/3ec95129139360c21be3f181724135daf4843579/schema-kenerator-reflection/src/main/kotlin/io/github/smiley4/schemakenerator/reflection/steps/ReflectionTypeProcessingStep.kt#L451-L459

For example, consider the following class:

```kotlin
class Request(
    val name: OptionalValue<String?>
)
```

The call to `resolveMemberType(member.returnType, resolvedTypeParameters, typeData).id` will apply the redirect to `String?`, but the response does not contain the information that `String?` is nullable. Then `nullable = member.returnType.isMarkedNullable` assigns `false`, because `Request.name` itself is not nullable, it is only redirected to a nullable type.

**Comment by SMILEY4** [comment]
Hi,
Thank you for reporting the issue and the hints. I created a fix and released version 1.2.0.  ktor-swagger-ui should receive a new version soon aswell.

**Comment by mnonnenmacher** [comment]
Thank you very much!
